### PR TITLE
Transform updates

### PR DESCRIPTION
**Pull request by 1chandu**
For #576 

#### Background
Adding changes to make `Transform` class ready for release.

#### Change List
- Deduce varyings when not provided.
- Add utility method to get version from shader string, use it in `Transform`
- Update docs.
- Add unit tests.

### CODE REVIEW

File: docs/api-reference/core/transform.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ Constructs a `Transform` object. It then creates destination buffers if needed a
   * `sourceBuffers` (`Object`) - key and value pairs, where key is the name of vertex shader attribute and value is the corresponding `Attribute`, `Buffer` or attribute descriptor object.
   * `feedbackBuffers` (`Object`, Optional) - key and value pairs, where key is the name of vertex shader attribute and value is the corresponding `Buffer` object.
   * `vs` (`String`) - vertex shader string.
-  * `varyings` (`Array`) - Array of vertex shader varyings names.
+  * `varyings` (`Array`) - Array of vertex shader varyings names. When not provided this can be deduced from `feedbackMap`. Either `varyings` or `feedbackMap` must be provided.
```
Comment (ibgreen) [inline docs/api-reference/core/transform.md]: In luma.gl v6 varyings are deduced automatically from the Model's program. Do we still need this?

File: src/webgl-utils/index.js [hunk 2]
```diff
@@ -37,3 +37,5 @@ export {
 } from './typed-array-utils';
 
 export {getKeyValue, getKey, getKeyType} from './constants-to-keys';
+
+export {getVersion} from './shader-utils';
```
Comment (ibgreen) [inline src/webgl-utils/index.js]: Name is a bit generic? `getShaderVersion`?

File: src/core/transform.js [hunk 3]
```diff
@@ -129,21 +132,28 @@ export default class Transform {
...
-    assert(sourceBuffers && vs && Array.isArray(varyings) && elementCount >= 0);
+    assert(sourceBuffers && vs && elementCount >= 0);
     // If feedbackBuffers are not provided, sourceDestinationMap must be provided
     // to create destinaitonBuffers with layout of corresponding source buffer.
     assert(feedbackBuffers || feedbackMap, ' Transform needs feedbackBuffers or feedbackMap');
     for (const bufferName in feedbackBuffers || {}) {
       assert(feedbackBuffers[bufferName] instanceof Buffer);
     }
 
+    // If varyings are not provided feedbackMap must be provided to build varyings
```
Comment (ibgreen) [inline src/core/transform.js]: Comment: build varyings => deduce varyings?

File: src/core/transform.js [hunk 4]
```diff
@@ -129,21 +132,28 @@ export default class Transform {
...
+    assert(Array.isArray(varyings) || feedbackMap);
+    let varyingsArray = varyings;
+    if (!Array.isArray(varyings)) {
+      varyingsArray = Object.values(feedbackMap);
```
Comment (ibgreen) [inline src/core/transform.js]: Nit: You could just do `varyings = Object.values...` and not declare a new var.

File: src/core/transform.js [hunk 5]
```diff
@@ -181,7 +191,7 @@ export default class Transform {
 
   // build Model and TransformFeedback objects
   _buildModel({id, vs, varyings, drawMode, elementCount}) {
-    const fs = this._getEmptyFragmentShader(vs);
+    const fs = getEmptyFragmentShader(vs);
```
Comment (ibgreen) [inline src/core/transform.js]: Nit: Maybe just add a ternary here and drop the function?

```
const fs = getShaderVersion(vs) ? FS300 : FS100;
```

File: docs/api-reference/core/transform.md (hunk 1 above)
Comment (1chandu) [reply docs/api-reference/core/transform.md]: Not all varyings defined in vertex shader have to be captured, ie. this varyings array could be subset of all varyings.

File: src/core/transform.js (hunk 4 above)
Comment (1chandu) [reply src/core/transform.js]: Just avoiding changing object passed to this method.

**Pull request merged**